packages/langium/src/grammar/validation/validator.ts

```diff
@@ -809,6 +814,108 @@ export class LangiumGrammarValidator {
         }
     }
 
+    /** This validation recursively looks at all assignments (and rewriting actions) with '=' as assignment operator and checks,
+     * whether the operator should be '+=' instead. */
+    checkOperatorMultiplicitiesForMultiAssignments(rule: ast.ParserRule, accept: ValidationAcceptor): void {
+        // for usual parser rules AND for fragments, but not for data type rules!
+        if (!rule.dataType) {
+            this.checkOperatorMultiplicitiesForMultiAssignmentsLogic([rule.definition], accept);
+        }
+    }
+
+    private checkOperatorMultiplicitiesForMultiAssignmentsLogic(startNodes: AstNode[], accept: ValidationAcceptor): void {
+        // new map to store usage information of the assignments
+        const map: Map<string, AssignmentUse> = new Map();
+
+        // top-down traversal for all starting nodes
+        for (const node of startNodes) {
+            this.checkAssignmentNumbersForNode(node, 1, map, accept);
+        }
+
+        // create the warnings
+        for (const entry of map.entries()) {
+            if (entry[1].counter >= 2) {
+                for (const assignment of entry[1].assignments) {
+                    if (assignment.operator !== '+=') {
+                        accept(
+                            'warning',
+                            `It seems, that you are assigning multiple values to the feature '${assignment.feature}', while you are using '${assignment.operator}' as assignment operator. Consider to use '+=' instead in order not to loose some of the assigned values.`,
+                            { node: assignment, property: 'operator' }
```

Comment:
Suggestion: I would rephrase that validation message to make it a bit more imperative. Also the info marker on the `operator` keyword is pretty hard to see. I would use the `feature` property instead.
```suggestion
                            `Found multiple assignments to '${assignment.feature}' with the '${assignment.operator}' assignment operator. Consider using '+=' instead to prevent data loss.`,
                            { node: assignment, property: 'feature' }
```

Reply:
Thank makes sense, changed